### PR TITLE
[LYFT][STRMCMP-601] Custom FlinkKafkaProducer translation

### PR DESCRIPTION
**Pull request by tweise**

### CODE REVIEW

File: runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java [hunk 1]
```diff
@@ -157,6 +163,45 @@ public boolean isEndOfStream(WindowedValue<byte[]> nextElement) {
     }
   }
 
+  // TODO: translation assumes byte[] values, does not support keys and headers
+  private void translateKafkaSink(
+          String id, RunnerApi.Pipeline pipeline, StreamingTranslationContext context) {
```
Comment (rakeshcusat) [inline runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java]: nit: can we make these `final` variables?

File: runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java [hunk 2]
```diff
@@ -70,6 +73,7 @@
       LoggerFactory.getLogger(LyftFlinkStreamingPortableTranslations.class.getName());
 
   private static final String FLINK_KAFKA_URN = "lyft:flinkKafkaInput";
+  private static final String FLINK_KAFKA_SINK_URN = "lyft:flinkKafkaSink";
```
Comment (rakeshcusat) [inline runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java:76]: `lyft:flinkKafkaSink` gets a little bit confusing. Can it be `lyft:flinkKafkaProducer`?

File: sdks/python/apache_beam/io/lyft/kafka.py [hunk 3]
```diff
@@ -58,3 +58,57 @@ def with_bootstrap_servers(self, bootstrap_servers):
...
     return self.set_kafka_consumer_property('group.id', group_id)
+
+
+class FlinkKafkaSink(PTransform):
+  """Custom transform that wraps a Flink Kafka producer - only works with the
+  portable Flink runner.
+
+  See:
+  https://ci.apache.org/projects/flink/flink-docs-stable/dev/connectors/kafka.html#kafka-producer
+
```
Comment (rakeshcusat) [inline sdks/python/apache_beam/io/lyft/kafka.py:77]: I would also add an example of a usage of this class, so people will not bug you in future.

**Comment by rakeshcusat** [comment]
Thanks for taking care of this.

**Comment by rakeshcusat** [comment]
@tweise I just noticed that it is based on `release-2.16.0-lyft` branch since there are performance regressions on 2.16, can we base this on `release-2.14.0-lyft` so we can test it out in production?

File: runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java [hunk 4]
```diff
@@ -53,11 +53,14 @@
...
 import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.Iterables;
 import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.api.common.serialization.SerializationSchema;
 import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.streaming.api.datastream.DataStream;
 import org.apache.flink.streaming.api.datastream.DataStreamSource;
 import org.apache.flink.streaming.api.functions.timestamps.BoundedOutOfOrdernessTimestampExtractor;
 import org.apache.flink.streaming.api.windowing.time.Time;
 import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer010;
+import org.apache.flink.streaming.connectors.kafka.FlinkKafkaProducer011;
```
Comment (rakeshcusat) [inline runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java:63]: should we also change `FlinkKafkaConsumer010` => `FlinkKafkaConsumer011` 
I can take care of this in https://github.com/lyft/beam/pull/20

Comment (tweise) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java:63]: Changed in this PR.

**Pull request merged**